cedar-policy-validator/src/schema.rs

```diff
@@ -3332,6 +3334,38 @@ pub(crate) mod test {
             }
         });
         assert_invalid_json_schema(src);
+
+        // Cedar examines common type name declarations eagerly.
+        // So it throws an error for the following example even though `Record`
+        // is not referenced.
+        let src: serde_json::Value = json!({
+            "": {
+                "commonTypes": {
+                    "Record": {
+                        "type": "Record",
+                        "attributes": {
+                            "a": {
+                                "type": "Long",
+                            }
+                        }
+                    }
+                },
+                "entityTypes": {
+                    "b": {
+                        "shape" : {
+                            "type" : "Record",
```

Comment:
It's referenced here?

Reply:
oh, entity attributes have a hack so that this doesn't count. Would still rather see the test using `Set` so future me isn't confused again

Reply:
Done?